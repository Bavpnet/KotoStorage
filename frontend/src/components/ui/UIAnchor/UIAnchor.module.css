.anc {
  background-color: black;
  color: white;
  border-radius: 10px;
  justify-content: center;
  align-items: center;
  display: flex;
  border: none;
  font-size: 1rem;
  font-weight: 400;
  padding: 8px 30px;
  text-decoration: none;
}
