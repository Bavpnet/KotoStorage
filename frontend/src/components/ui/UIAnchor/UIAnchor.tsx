import styles from './UIAnchor.module.css'

type UIAnchorProps = {
  text: string
}

export const UIAnchor: React.FC<
  React.HTMLProps<HTMLAnchorElement> & UIAnchorProps
> = ({ text, ...rest }) => (
  <a className={styles.anc} {...rest}>
    {text}
  </a>
)
