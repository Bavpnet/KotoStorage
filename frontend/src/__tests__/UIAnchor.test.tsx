import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import renderer from 'react-test-renderer'
import { UIAnchor } from '../components/ui/UIAnchor'

describe('UIAnchor', () => {
  it('renders correctly', () => {
    const text = 'test'
    const tree = renderer.create(<UIAnchor text={text} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with href', () => {
    const text = 'test'
    const href = 'https://www.google.com/'
    const { getByRole } = render(<UIAnchor text={text} href={href} />)

    const anchor = getByRole('link')

    expect(anchor).toHaveTextContent(text)
    expect(anchor).toHaveAttribute('href', href)
  })
})
