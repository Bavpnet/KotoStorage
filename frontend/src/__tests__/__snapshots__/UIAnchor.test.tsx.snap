// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UIAnchor renders correctly 1`] = `
<a
  className="anc"
>
  test
</a>
`;
